### ISSUE TITLE
PBKDF fails when salt is multiple of 4 bytes.

### ISSUE THREAD

**Issue by trounce1** [bkerler/opencl_brute#7]
change the salt in test.py to a multiple of 4 bytes (e.g. b'salty123' or b'salt') and it has a weird output pattern where every calculated hash is incorrect.

for a password list of password 10 times then the calculated hashes will be

hash0
hash1
hash1
hash1
hash1
hash1
hash1
hash1
hash0
hash1

hash0 repeats every 8 and is incorrect and hash1 is also incorrect.
this is the behaviour for sha1 and sha256

for md5 it repeats every 16 instead

all other tests pass

**Comment by bkerler** [comment]
Can you show me some example values ? Also which gpu and which opencl runtime are you using ?

**Comment by trounce1** [comment]
HOSTs: 1
	Apple  -  FULL_PROFILE  -  OpenCL 1.2 (Jun 23 2019 21:50:55)
CPUs:  1
	Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz , from  Intel

GPUs:  2  (OpenCL Enabled)
GPU 1 : Intel(R) HD Graphics 630 , from  Intel Inc.
Specs:
	Local Memory Size:		65536
	Max Compute Units:		24
	MAX Work Item Dimentions:	3  [ 256 | 256 | 256 ]
	Work Group Size:		256
	Global Memory Size:		1 GigaBytes
	Constant Buffer Size:		65536 Bytes
	Device Profile:			Device Supports FULL OpenCL specification	( OpenCL version: 1.2(Jul  2 2019 21:50:31) )
GPU 2 : AMD Radeon Pro 560 Compute Engine , from  AMD
Specs:
	Local Memory Size:		32768
	Max Compute Units:		16
	MAX Work Item Dimentions:	3  [ 256 | 256 | 256 ]
	Work Group Size:		256
	Global Memory Size:		4 GigaBytes
	Constant Buffer Size:		65536 Bytes
	Device Profile:			Device Supports FULL OpenCL specification	( OpenCL version: 1.2 (Jul  2 2019 21:41:38) )


passwords : password, pass, batman
salt : salty123
Testing pbkdf2-hmac using md5.cl
Failed !
0
b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf m6\x94W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89jW\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j\'\xfa'
b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf m6\x94\xdb\xc9\rA\t\xe1\xc4\xa6s\x8bp\xe0\xfa\x0b\xd2\xe6W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j|\xe4'
1
b'\xf3\xf7\xce\xb0\xfdP*,"s1\xaf]\xcbb\x02\xfa\xa8\xde\x945S\xfe\x8e\x1ah\xa95U\n\x08WLg"\xdcS\x80\x02S5\xf4\x99f\xf5>H\xd7\xbb\x8a'
b'w\xd0\xfc\xd8\xb9\xb7\xba\x86\x9a\xe3_\x9f\x7f8\xe3\x15{\xaa\x8a0`\x1el\xe2\xc5\xbf\x97\x1d~\xd5l\xbd\xe3\x8a\xc1urQ1\x9f\xe3\n\xc4\xf1$;\xa7\xf5\xc2g'
2
b'I\xa0\xc8\x08&\xcf\xf6\xeb]h|W\xf3*\xdb\x17\\\xabG\xf9H\xef\xcc\xfc7\x91\xa9aCw3\x99\xe9\x16"?\x9bH\xa7\xaf\xe5\xaf3\xcd\xbf2\xf7\xb2\x1br'
b'_/\x044\xa8\xb3L\xb7\x12\xc6\xaa\xa53\x1c\xd1_%\xdd5\x99\x12gt\x9a\x01\n\x9b\x7f\x94\x96o\x86(\xd8\x9e\xec\xe4Z\x90c_9\x8a\xc2\x9ct\xfc\xa4"\xfa'
Testing pbkdf2-hmac using sha1.cl
Failed !
0
b"\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81P\x88\xf6\x0f\x861\xf0\x8a[\xd9\\\xca\xcf\x0b\xfb'\x85s\r\x8e\xd0\x88\xf6\x0f\x861\xf0\x8a[\xd9\\"
b'\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81Ppszr"\xb5\x8b\x1f(\x19\xaa\xf0c\xc7c^jA\x9bb\x88\xf6\x0f\x861\xf0\x8a[\xd9\\'
1
b'\x94\xa2d\x8bW\xfe5\xfc\xbb\x98?`E\x12&y.Q\xa2\xb1$]\xd5\x90I\xae\xac}\x8b3jB\x9b\x01\xd3\xef?sRD\xb6\x90*Xa\x98\xd5o\xb9{'
b'\xe7G8\x7f\xca\xbd\x92_\te\xfa\xabq\x80;W\xe3\xb9\x90\xb9uK\xec\xde\xd4\xc18\xc4\xa9K\xcc#6k\xd6\xe1\xaa\x17\xc1\xae\x1dJWhN\xe4\xff\xdc6\xa0'
2
b'\xc2 9\n\xde\xa0s*\xeb\xc6E\xf3%d\xd2\x12E\xeb\xed>W|\x89\xe3l\x9f\x8dBV\xf5\xaa\xb2 \xf9{\x8c\xce\xeeA\xb8\x7fk\xe7\xff\xbc@\xd1\xf7\x828'
b'\xd8`\xc3\xef\xd8u\xe0\x99\xb7\xd5\x9a\xf2l\xddg]\xab\xa8\xdf\x95\xf5\xe4\xa9\xf0\x91\xc7\xa2\x19\xcb\xe3\x8f\x0b\xc4T\x1d+\x1c\x12\x08\x828\x01\xe9\x13\xef\x88\xbd\x94Sn'
Tests have finished.

**Comment by bkerler** [comment]
yeah, that's why I asked. amd have a memory issue and do not handle some values in opencl properly. as amd also no longer officially supports opencl when I last checked (no updated sdk), we currently do all our tests on Nvidia because of that. if I get my hands on an amd, I will try to fix it, but no promises on that.



Am 30. September 2019 12:53:59 MESZ schrieb TROUNCE <notifications@github.com>:
>HOSTs: 1
>	Apple  -  FULL_PROFILE  -  OpenCL 1.2 (Jun 23 2019 21:50:55)
>CPUs:  1
>	Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz , from  Intel
>
>GPUs:  2  (OpenCL Enabled)
>GPU 1 : Intel(R) HD Graphics 630 , from  Intel Inc.
>Specs:
>	Local Memory Size:		65536
>	Max Compute Units:		24
>	MAX Work Item Dimentions:	3  [ 256 | 256 | 256 ]
>	Work Group Size:		256
>	Global Memory Size:		1 GigaBytes
>	Constant Buffer Size:		65536 Bytes
>	Device Profile:			Device Supports FULL OpenCL specification	( OpenCL
>version: 1.2(Jul  2 2019 21:50:31) )
>GPU 2 : AMD Radeon Pro 560 Compute Engine , from  AMD
>Specs:
>	Local Memory Size:		32768
>	Max Compute Units:		16
>	MAX Work Item Dimentions:	3  [ 256 | 256 | 256 ]
>	Work Group Size:		256
>	Global Memory Size:		4 GigaBytes
>	Constant Buffer Size:		65536 Bytes
>	Device Profile:			Device Supports FULL OpenCL specification	( OpenCL
>version: 1.2 (Jul  2 2019 21:41:38) )
>
>
>passwords : password, pass, batman
>salt : salty123
>Testing pbkdf2-hmac using md5.cl
>Failed !
>0
>b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf
>m6\x94W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89jW\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j\'\xfa'
>b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf
>m6\x94\xdb\xc9\rA\t\xe1\xc4\xa6s\x8bp\xe0\xfa\x0b\xd2\xe6W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j|\xe4'
>1
>b'\xf3\xf7\xce\xb0\xfdP*,"s1\xaf]\xcbb\x02\xfa\xa8\xde\x945S\xfe\x8e\x1ah\xa95U\n\x08WLg"\xdcS\x80\x02S5\xf4\x99f\xf5>H\xd7\xbb\x8a'
>b'w\xd0\xfc\xd8\xb9\xb7\xba\x86\x9a\xe3_\x9f\x7f8\xe3\x15{\xaa\x8a0`\x1el\xe2\xc5\xbf\x97\x1d~\xd5l\xbd\xe3\x8a\xc1urQ1\x9f\xe3\n\xc4\xf1$;\xa7\xf5\xc2g'
>2
>b'I\xa0\xc8\x08&\xcf\xf6\xeb]h|W\xf3*\xdb\x17\\\xabG\xf9H\xef\xcc\xfc7\x91\xa9aCw3\x99\xe9\x16"?\x9bH\xa7\xaf\xe5\xaf3\xcd\xbf2\xf7\xb2\x1br'
>b'_/\x044\xa8\xb3L\xb7\x12\xc6\xaa\xa53\x1c\xd1_%\xdd5\x99\x12gt\x9a\x01\n\x9b\x7f\x94\x96o\x86(\xd8\x9e\xec\xe4Z\x90c_9\x8a\xc2\x9ct\xfc\xa4"\xfa'
>Testing pbkdf2-hmac using sha1.cl
>Failed !
>0
>b"\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81P\x88\xf6\x0f\x861\xf0\x8a[\xd9\\\xca\xcf\x0b\xfb'\x85s\r\x8e\xd0\x88\xf6\x0f\x861\xf0\x8a[\xd9\\"
>b'\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81Ppszr"\xb5\x8b\x1f(\x19\xaa\xf0c\xc7c^jA\x9bb\x88\xf6\x0f\x861\xf0\x8a[\xd9\\'
>1
>b'\x94\xa2d\x8bW\xfe5\xfc\xbb\x98?`E\x12&y.Q\xa2\xb1$]\xd5\x90I\xae\xac}\x8b3jB\x9b\x01\xd3\xef?sRD\xb6\x90*Xa\x98\xd5o\xb9{'
>b'\xe7G8\x7f\xca\xbd\x92_\te\xfa\xabq\x80;W\xe3\xb9\x90\xb9uK\xec\xde\xd4\xc18\xc4\xa9K\xcc#6k\xd6\xe1\xaa\x17\xc1\xae\x1dJWhN\xe4\xff\xdc6\xa0'
>2
>b'\xc2
>9\n\xde\xa0s*\xeb\xc6E\xf3%d\xd2\x12E\xeb\xed>W|\x89\xe3l\x9f\x8dBV\xf5\xaa\xb2
>\xf9{\x8c\xce\xeeA\xb8\x7fk\xe7\xff\xbc@\xd1\xf7\x828'
>b'\xd8`\xc3\xef\xd8u\xe0\x99\xb7\xd5\x9a\xf2l\xddg]\xab\xa8\xdf\x95\xf5\xe4\xa9\xf0\x91\xc7\xa2\x19\xcb\xe3\x8f\x0b\xc4T\x1d+\x1c\x12\x08\x828\x01\xe9\x13\xef\x88\xbd\x94Sn'
>Tests have finished.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/bkerler/opencl_brute/issues/7#issuecomment-536509942

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by trounce1** [comment]
no issue, i can develop for nvidia too. thanks

**Comment by JJJollyjim** [comment]
I have exactly the same issue on an nvidia card.

Simply changing 'salty' to 'salty123', I get the following output:

```
Testing md5 ..
Ok m8!
Testing sha256 ..
Ok m8!
Testing sha1 ..
Ok m8!
Testing hmac using md5.cl
Ok m9!
Testing hmac using sha1.cl
Ok m9!
Testing hmac using sha256.cl
Ok m9!
Testing pbkdf2-hmac using md5.cl
Failed !
0
b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf m6\x94W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89jW\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j\'\xfa'
b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf m6\x94\xdb\xc9\rA\t\xe1\xc4\xa6s\x8bp\xe0\xfa\x0b\xd2\xe6W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j|\xe4'
1
b"\xf5YU\xe1\x7f_\xe5\xc3\x19\xf8F\xb1\xc0\tRA\xd4\xb4\xaebEY\xfbr'\xed\xe2=\x93iA \xd4\xb4\xaebEY\xfbr'\xed\xe2=\x93iA Q."
b"\xf5YU\xe1\x7f_\xe5\xc3\x19\xf8F\xb1\xc0\tRAcQ\x805\x00\xe1\x99\x0f\xee\xff\x87\x10.l\x93\x7f\xd4\xb4\xaebEY\xfbr'\xed\xe2=\x93iA ZT"
2
b'D\x9eVGSfb5\xf2\xb0\xa8MKn\xda\x92\xdd\xab\xe2\x80\x93\xe1\xb3\xcc\x8do\xd9%\xbd\xf3\xd7\x11\xdd\xab\xe2\x80\x93\xe1\xb3\xcc\x8do\xd9%\xbd\xf3\xd7\x11\xb3\xd7'
b'D\x9eVGSfb5\xf2\xb0\xa8MKn\xda\x92\x1fV\xad\xc9\x94l>\xb9U\x9b\x8e\\2\xc0\x01T\xdd\xab\xe2\x80\x93\xe1\xb3\xcc\x8do\xd9%\xbd\xf3\xd7\x114B'
3
b'V#\x08=\xc1\xb8`W\x9a\x86_\xb1\xd8\xd6\xdd\xd7\xec\xba\xfaX\np$\xd1\xf5s~EvH8L\xec\xba\xfaX\np$\xd1\xf5s~EvH8L\xe1\x1b'
b'V#\x08=\xc1\xb8`W\x9a\x86_\xb1\xd8\xd6\xdd\xd7\xd5\xca\x80\x03=\xb5j\x1d\x12M\x15\x98\xd0\xddT\x8c\xec\xba\xfaX\np$\xd1\xf5s~EvH8L\xeb\xe2'
Testing pbkdf2-hmac using sha1.cl
Failed !
0
b"\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81P\x88\xf6\x0f\x861\xf0\x8a[\xd9\\\xca\xcf\x0b\xfb'\x85s\r\x8e\xd0\x88\xf6\x0f\x861\xf0\x8a[\xd9\\"
b'\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81Ppszr"\xb5\x8b\x1f(\x19\xaa\xf0c\xc7c^jA\x9bb\x88\xf6\x0f\x861\xf0\x8a[\xd9\\'
1
b'$\x00\x95\x90\xe2E\xdd\xdap\xe9f\xd8k\xd0\x9d\xafY\x1c\xf7\x18\xb1}1\xb7\xf6\x1b\xfe\xf3\x0ca\x10\x01zw\xa3\xf3\x06\xb8\xc0\xc7\xb1}1\xb7\xf6\x1b\xfe\xf3\x0ca'
b'$\x00\x95\x90\xe2E\xdd\xdap\xe9f\xd8k\xd0\x9d\xafY\x1c\xf7\x18\x02\xe1\x08\xacz\xe3B\x1a\x84\x1bcGL\xc2\x84\xbf\x1bd0\xd9\xb1}1\xb7\xf6\x1b\xfe\xf3\x0ca'
2
b'4\xa5\xe6\xdb\x94\xa6\xf2(S\x89\xe2\xf6\xbb\xc7\xa7i4\x06\xf1\x10\x92t;/\x9bto\x9eK#\xc4+\x1b\xc3(\x00\xf1\x9fL\xd5\x92t;/\x9bto\x9eK#'
b'4\xa5\xe6\xdb\x94\xa6\xf2(S\x89\xe2\xf6\xbb\xc7\xa7i4\x06\xf1\x10\xaaS\xe8`\x03\xb5N\x1c\x87R\x9f\x06\xa6\x1f\xd9\xe77\xdc>\xd7\x92t;/\x9bto\x9eK#'
3
b'Q\xa7QW\xae\x01\x8b=\x0e\x90\x0c\x16Gg\xcc\xc5\x0e\xb2x\x87cH\xdb\x8aE\xabk\xf8\xb6\x0f\x12Z\xb6x\xb2\x80\x1f}\xb9:cH\xdb\x8aE\xabk\xf8\xb6\x0f'
b'Q\xa7QW\xae\x01\x8b=\x0e\x90\x0c\x16Gg\xcc\xc5\x0e\xb2x\x87\x93H(\xb5\xbf\x01\xe4g2\xa0\xe2&4P\x1b\xb1\xda\xb9\xe3/cH\xdb\x8aE\xabk\xf8\xb6\x0f'
Testing pbkdf2-hmac using sha256.cl
Failed !
0
b"'\x8a\x8dpw?\x8a\xd0\xc9\x03\xd7\x96\xc5\xfd\xeeH\xcf\x9f?=,\x90\xbeKm\x8f\xa1\x0f\xa6\xa5\x04@ ?\xac\x0e\xdb\xe5L\xabVM \r\xe9\xf1\xa5//\xf6"
b"'\x8a\x8dpw?\x8a\xd0\xc9\x03\xd7\x96\xc5\xfd\xeeH\xcf\x9f?=,\x90\xbeKm\x8f\xa1\x0f\xa6\xa5\x04@\xd2\x1f\x0e\xac\x1b;yR\x18@!O\xc0v\xb9\x04f\x92"
1
b'\xedZ\x88\xb4\x16\x07\xd8~A\x17<\xb9\t)\xea\x93y\xa0\xddv\xe5\xbd\x8a\x80t\xf2\xdf\x89|\x96\t{\x04\xba[\x87%\xde\x11g\x90\xc7?3m\xd3\x83\x95~\xd9'
b'\xedZ\x88\xb4\x16\x07\xd8~A\x17<\xb9\t)\xea\x93y\xa0\xddv\xe5\xbd\x8a\x80t\xf2\xdf\x89|\x96\t{\xd2\xadu\x9e+\x08\x99T\xc7{\xbaJ\xdd\x88\xfb\x9f\xfc\xdb'
2
b'\xb7\xd3\xc7\xd62\x0b\xa0\xed\xe5\xa6p\x02Na\x84Wos6\xb4Bw\xe9\x9e=\x05\xe2/<,\x9f\xd9<K\x91\xf0\xcc(6i\xcb\x9d\xba\xee\xa5\x18*Cm{'
b"\xb7\xd3\xc7\xd62\x0b\xa0\xed\xe5\xa6p\x02Na\x84Wos6\xb4Bw\xe9\x9e=\x05\xe2/<,\x9f\xd9\xa2-\xb8\xd6,\xde\xc7\x90\xa8\xc2y'(\xee,\xcc\x8c\xdc"
3
b"3F/\xc4-\xab\xb1\x95\xda\x89\x06\xbc\xddQ\x1b\xe5]\xa4v\xba:\x19&\x0ccl(Z\xa74\x03\xae\xef\xd9\xee\xaa'\xfa7'c\x90\xe1\x9f\x15#\x92\x82\x120"
b'3F/\xc4-\xab\xb1\x95\xda\x89\x06\xbc\xddQ\x1b\xe5]\xa4v\xba:\x19&\x0ccl(Z\xa74\x03\xae\xc7AY\xed}\xdc\x04\x9fZ\xf9\x88;r\xab\x82\xee0\x03'
Testing scrypt
Ok m11!
Tests have finished.
```

Hardware details:

```
Platform 0 - Name NVIDIA CUDA, Vendor NVIDIA Corporation

============================================================
OpenCL Platforms and Devices
============================================================
Platform 0 - Name: NVIDIA CUDA
Platform 0 - Vendor: NVIDIA Corporation
Platform 0 - Version: OpenCL 1.2 CUDA 10.1.0
Platform 0 - Profile: FULL_PROFILE
 --------------------------------------------------------
 Device - Name: GeForce GTX 970
 Device - Type: GPU
 Device - Max Clock Speed: 1367 Mhz
 Device - Compute Units: 13
 Device - Local Memory: 48 KB
 Device - Constant Memory: 64 KB
 Device - Global Memory: 4 GB
 Device - Max Buffer/Image Size: 1010 MB
 Device - Max Work Group Size: 1024
```

**Comment by JJJollyjim** [comment]
Also the same problem on intel:

```
Testing md5 ..
Ok m8!
Testing sha256 ..
Ok m8!
Testing sha1 ..
Ok m8!
Testing hmac using md5.cl
Ok m9!
Testing hmac using sha1.cl
Ok m9!
Testing hmac using sha256.cl
Ok m9!
Testing pbkdf2-hmac using md5.cl
Failed !
0
b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf m6\x94W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89jW\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j\'\xfa'
b'9O\xfe\xc6/\x1e;?\x1a\xe89\xdf m6\x94\xdb\xc9\rA\t\xe1\xc4\xa6s\x8bp\xe0\xfa\x0b\xd2\xe6W\x0b\xf9e\xf7\xf2)8D\x99,"\\\x92\x89j|\xe4'
1
b"\xf5YU\xe1\x7f_\xe5\xc3\x19\xf8F\xb1\xc0\tRA\xd4\xb4\xaebEY\xfbr'\xed\xe2=\x93iA \xd4\xb4\xaebEY\xfbr'\xed\xe2=\x93iA Q."
b"\xf5YU\xe1\x7f_\xe5\xc3\x19\xf8F\xb1\xc0\tRAcQ\x805\x00\xe1\x99\x0f\xee\xff\x87\x10.l\x93\x7f\xd4\xb4\xaebEY\xfbr'\xed\xe2=\x93iA ZT"
2
b'D\x9eVGSfb5\xf2\xb0\xa8MKn\xda\x92\xdd\xab\xe2\x80\x93\xe1\xb3\xcc\x8do\xd9%\xbd\xf3\xd7\x11\xdd\xab\xe2\x80\x93\xe1\xb3\xcc\x8do\xd9%\xbd\xf3\xd7\x11\xb3\xd7'
b'D\x9eVGSfb5\xf2\xb0\xa8MKn\xda\x92\x1fV\xad\xc9\x94l>\xb9U\x9b\x8e\\2\xc0\x01T\xdd\xab\xe2\x80\x93\xe1\xb3\xcc\x8do\xd9%\xbd\xf3\xd7\x114B'
3
b'V#\x08=\xc1\xb8`W\x9a\x86_\xb1\xd8\xd6\xdd\xd7\xec\xba\xfaX\np$\xd1\xf5s~EvH8L\xec\xba\xfaX\np$\xd1\xf5s~EvH8L\xe1\x1b'
b'V#\x08=\xc1\xb8`W\x9a\x86_\xb1\xd8\xd6\xdd\xd7\xd5\xca\x80\x03=\xb5j\x1d\x12M\x15\x98\xd0\xddT\x8c\xec\xba\xfaX\np$\xd1\xf5s~EvH8L\xeb\xe2'
Testing pbkdf2-hmac using sha1.cl
Failed !
0
b'\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81Pe\xd5\xee\xc7<\x81`\x9c\xd4ml0\x1c\\\x07\xa8\x10[\x8c\x81\x95\x8c>L]9\xb0\x9b?m'
b'\x81\xd2\x7f[\xa0E\xff\xc52/\x15\xce\x97Lc\xcd{\xc1\x81Ppszr"\xb5\x8b\x1f(\x19\xaa\xf0c\xc7c^jA\x9bb\x88\xf6\x0f\x861\xf0\x8a[\xd9\\'
1
b'$\x00\x95\x90\xe2E\xdd\xdap\xe9f\xd8k\xd0\x9d\xafY\x1c\xf7\x18\xea+m \xae$\x11\x0c\x81\xc5;\xf3\xdcd\xfc\xe0$\x9810\xe6\xbc\x88\xd4\xf1$\xfd\x8dI\xc1'
b'$\x00\x95\x90\xe2E\xdd\xdap\xe9f\xd8k\xd0\x9d\xafY\x1c\xf7\x18\x02\xe1\x08\xacz\xe3B\x1a\x84\x1bcGL\xc2\x84\xbf\x1bd0\xd9\xb1}1\xb7\xf6\x1b\xfe\xf3\x0ca'
2
b'4\xa5\xe6\xdb\x94\xa6\xf2(S\x89\xe2\xf6\xbb\xc7\xa7i4\x06\xf1\x10a\xe7m\x0f\x97\xfbDH\xb1\xb8I\xc0bL\x0bB\xea\xe0`~\nw0\xe0\xbe\x0f\xf8\x8c\x9b\x86'
b'4\xa5\xe6\xdb\x94\xa6\xf2(S\x89\xe2\xf6\xbb\xc7\xa7i4\x06\xf1\x10\xaaS\xe8`\x03\xb5N\x1c\x87R\x9f\x06\xa6\x1f\xd9\xe77\xdc>\xd7\x92t;/\x9bto\x9eK#'
3
b'Q\xa7QW\xae\x01\x8b=\x0e\x90\x0c\x16Gg\xcc\xc5\x0e\xb2x\x87\xb4\xac%)\xf6L\x80C\x13\xae\x8f\xfd$\xaf\xeaEIp\x04\xf3\xdf~v\xe5\x88\xda \x82\xbd\x87'
b'Q\xa7QW\xae\x01\x8b=\x0e\x90\x0c\x16Gg\xcc\xc5\x0e\xb2x\x87\x93H(\xb5\xbf\x01\xe4g2\xa0\xe2&4P\x1b\xb1\xda\xb9\xe3/cH\xdb\x8aE\xabk\xf8\xb6\x0f'
Testing pbkdf2-hmac using sha256.cl
Failed !
0
b"'\x8a\x8dpw?\x8a\xd0\xc9\x03\xd7\x96\xc5\xfd\xeeH\xcf\x9f?=,\x90\xbeKm\x8f\xa1\x0f\xa6\xa5\x04@\xedN\xfd\xae\x0e\xed\xc4\x9dC\x14<\x95\x16\x92\x8b\x99Q\x8e"
b"'\x8a\x8dpw?\x8a\xd0\xc9\x03\xd7\x96\xc5\xfd\xeeH\xcf\x9f?=,\x90\xbeKm\x8f\xa1\x0f\xa6\xa5\x04@\xd2\x1f\x0e\xac\x1b;yR\x18@!O\xc0v\xb9\x04f\x92"
1
b'\xedZ\x88\xb4\x16\x07\xd8~A\x17<\xb9\t)\xea\x93y\xa0\xddv\xe5\xbd\x8a\x80t\xf2\xdf\x89|\x96\t{\xc7\xa4\x04\xa2`\xe8\xc0\xb8~$\x80\r\xf9\xd4\xda\x82,\xb2'
b'\xedZ\x88\xb4\x16\x07\xd8~A\x17<\xb9\t)\xea\x93y\xa0\xddv\xe5\xbd\x8a\x80t\xf2\xdf\x89|\x96\t{\xd2\xadu\x9e+\x08\x99T\xc7{\xbaJ\xdd\x88\xfb\x9f\xfc\xdb'
2
b'\xb7\xd3\xc7\xd62\x0b\xa0\xed\xe5\xa6p\x02Na\x84Wos6\xb4Bw\xe9\x9e=\x05\xe2/<,\x9f\xd9\xf1`\x1d\x1f\xcb\xb6\x0e\x19\xae\x1fD\xc7\x84\x04M\xf4\xf5\xe9'
b"\xb7\xd3\xc7\xd62\x0b\xa0\xed\xe5\xa6p\x02Na\x84Wos6\xb4Bw\xe9\x9e=\x05\xe2/<,\x9f\xd9\xa2-\xb8\xd6,\xde\xc7\x90\xa8\xc2y'(\xee,\xcc\x8c\xdc"
3
b'3F/\xc4-\xab\xb1\x95\xda\x89\x06\xbc\xddQ\x1b\xe5]\xa4v\xba:\x19&\x0ccl(Z\xa74\x03\xaej\x03\xfc,\x14\xa78o\x0fG\xf0\xb7\xd2\xcc,\xff\xb4_'
b'3F/\xc4-\xab\xb1\x95\xda\x89\x06\xbc\xddQ\x1b\xe5]\xa4v\xba:\x19&\x0ccl(Z\xa74\x03\xae\xc7AY\xed}\xdc\x04\x9fZ\xf9\x88;r\xab\x82\xee0\x03'
```

Hardware info:

```
============================================================
OpenCL Platforms and Devices
============================================================
Platform 0 - Name: Intel(R) OpenCL HD Graphics
Platform 0 - Vendor: Intel(R) Corporation
Platform 0 - Version: OpenCL 2.1 
Platform 0 - Profile: FULL_PROFILE
 --------------------------------------------------------
 Device - Name: Intel(R) Gen9 HD Graphics NEO
 Device - Type: GPU
 Device - Max Clock Speed: 1100 Mhz
 Device - Compute Units: 24
 Device - Local Memory: 64 KB
 Device - Constant Memory: 3159716 KB
 Device - Global Memory: 6 GB
 Device - Max Buffer/Image Size: 3086 MB
 Device - Max Work Group Size: 256
```

**Comment by bkerler** [comment]
Confirmed. I will have a look what might cause this.

**Comment by bkerler** [comment]
Should be fixed in commit f22ae10. Please verify for both amd and nvidia if it works or what fails for which card. Thanks :)

**Comment by trounce1** [comment]
works a treat :)
